msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,322 @@
 
-#### Welcome messages in the team or channel  scope
+### 9.3 In-meeting tabs
 
-Things are a little bit different when the bot is first added to a channel. Normally, you shouldn't send a 1:1 message to everyone on the team, but the bot can send a welcome message in the channel.
+* Must be responsive and adapt within the provided space. Make sure to maintain padding and component sizes.
+* Must have a back button if there is more than one layer of navigation.
+* Must not include more than one dismiss or close button. This may confuse users since there's already a built-in header button in the header to dismiss the tab.
+* Must not have horizontal scrolling.
 
-### &#9989; Mobile responsiveness, no direct upsell or payment
+### 9.4 In-meeting dialogs
 
-* Your tabs, adaptive cards, bot messages and content in task modules must be responsive for a variety of mobile device screens sizes.
-* Apps that support iOS must be fully functional on the latest iPad device using the latest version of iOS.
-* Must not include any direct references to in-app purchases, trial offers, offers for paid versions, or links to any online stores where users can purchase or acquire other content, apps, or add-ins from your Teams app on mobile OS (Android, iOS).
-* The iOS or Android version of the add-in must not show any UI or language or link to any other apps, add-ins, or website that ask the user to pay.
-* The associated Privacy Policy and Terms of Use pages must also be free of any commerce UI or Store links.
+* Should be used sparingly and for scenarios that are light and task-oriented.
+* Must display content in a single column and not have multiple navigation levels.
+* Must not use task modules.
+* Should align with the center of the meeting stage.
+* Should be dismissed once a user selects a button or performs an action.
 
-### &#9989; Do not post sensitive data to an audience not intended to view the data
+## 10.0 Notification design
 
-Your Teams app must not post sensitive data such as credit card or financial payment instrument, Personal Identifiable Information (PIN), health, or contact tracing information to an audience not intended to view that data.
+Follow these guidelines if your app uses the [activity feed APIs provided by Microsoft Graph](https://docs.microsoft.com/graph/teams-send-activityfeednotifications).
 
-### &#9989; Do not transmit financial payment details or complete financial transactions via your Teams app
+### 10.1 General
 
-* Your Teams app must not ask users to make a payment directly within Teams interface.
-* Apps may not transmit financial instrument details through the user on the app interface. Apps may only transmit links to secure payment services to users if this is disclosed in the app's Terms of Use, Privacy Policy, and any profile page or website for the app before a user agrees to use the app.
+* All the notification triggers specified in your app manifest should get a notification in the app.
```

Comment:
I am not sure I follow. Are there situations where notifications specified in the manifest would not display in the Teams client?

Reply:
There can be. We are quoting a test case - everything mentioned in Manifest should be functional & result into right notification

Reply:
Got it! Thanks for clarifying.